UniversityLibrary/Teacher.cs

```diff
@@ -0,0 +1,35 @@
+﻿
+namespace UniversityLibrary
+{
+    public class Teacher : Human
+    {
+        public Teacher(string firstName, string lastName, int age, TeacherCategory teacherCategory) : base(firstName, lastName, age)
+        {
+            _teacherCategory = teacherCategory;
+        }
+
+        private TeacherCategory _teacherCategory;
+
+        public TeacherCategory TeacherCategory
+        {
+            get => _teacherCategory;
+            set
+            {
+                if (teacherCategoryValid((int)value))
+                    _teacherCategory = value;
+            }
+        }
+
+        public int CountOfStudentsInGroup { get; }
+
+        private bool teacherCategoryValid(int teacherCategory)
+        {
+            return teacherCategory > 0 && teacherCategory < 4;
```

Comment:
not use magic numbers
whar are you doing if your enum will be increased/decreased?